docs/dev-implementation.md

```diff
@@ -0,0 +1,124 @@
+# Implementation details
+
+This document provides an overview on how the `conda-libmamba-solver` integrations are implemented,
+both within the `conda_libmamba_solver` package itself, and as a `conda` plugin.
+
+## Repository structure
+
+* `.github/workflows/`: CI pipelines to run unit and upstream tests, as well as linting and performance benchmarks.
+  Some extra workflows might be added by the `conda/infra` settings.
+* `conda_libmamba_solver/`: The Python package. Check sections below for details.
+* `recipe/`: The conda-build recipe used for the PR build previews. It should be kept in sync with `conda-forge` and `defaults`.
+* `dev/`: Supporting scripts and configuration files to set up development environments.
+* `docs/`: Documentation sources.
+* `tests/`: PyTest testing infrastructure.
+* `pyproject.toml`: Project metadata. See below for details.
+
+## Project metadata
+
+The `pyproject.toml` file stores the required packaging metadata, 
+as well as some the configuration for some tools (`black`, `pytest`, etc.).
+
+Some peculiarities:
+
+* `flit_core` is the chosen backend for the packaging (as opposed to `setuptools`).
+* The `version` is dynamically taken from `conda_libmamba_solver/__init__.py`.
+* `black` uses a line length of 99 characters.
+* PyTest configurations are extensive but mainly to deal with upstream testing. 
```

Comment:
```suggestion
* PyTest configurations are extensive but mainly only deal with upstream testing. 
```
...maybe this wording is not totally correct, just trying to clarify what this bullet point is saying

Reply:
I was trying to say that the pytest configs in pyproject.toml are quite detailed, but they are only needed when we are running the upstream tests. They don't have much of an effect for `conda-libmamba-solver` itself.

Reply:
Gotcha! Ok so maybe this bullet point can be re-worded like: 
```suggestion
* PyTest configurations are extensive but are only necessary when dealing with upstream testing. 
```
What do you think?